### PR TITLE
[bitnami/rabbitmq] Set advanced configuration from an existing secret

### LINKED ISSUE #14017: [rabbitmq chart] Add an option to override advancedConfiguration with a pre existing secret
### Name and Version

bitnami/rabbitmq

### What is the problem this feature will solve?

The "advancedConfiguration" field could have sensitive data (Azure active directory configuration in my case) or be different between environments.

### What is the feature you are proposing to solve the problem?

My proposal is that it would be nice to have a way to load that configuration from a secret. Then we can address both issues, not expose sensitive data and have configurations per environment easily.

If needed I am willing to help, but I don't know where to start so I appreciate any help.

### What alternatives have you considered?

I think the helm post render option could solve this, but doesn't fit my needs because argocd isn't compatible with that helm's feature. And I think a native solution would be better to not add that complexity to the deploy

### PR DESCRIPTION
**Pull request by FranAguiar**
### Description of the change
Set advanced configuration from an existing secret instead of plain text in values yaml. Plain text still available

### Benefits
Loading advanced configuration from a secret allow to not expose sensitive data or have different configurations between different instances with the same chart.

### Possible drawbacks
If both option are set, the secret will be ignored.

### Applicable issues
  - fixes #14017 

### Checklist
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/rabbitmq/templates/statefulset.yaml [hunk 1]
```diff
@@ -360,15 +360,14 @@ spec:
...
-                path: advanced.config
-              {{- end }}
+          projected:
+            sources:
+              - secret:
+                  name: {{ printf "%s-config" (include "common.names.fullname" .) }}
+            {{- if and .Values.advancedConfigurationExistingSecret (not .Values.advancedConfiguration) }}
+            - secret:
+                name: {{ tpl .Values.advancedConfigurationExistingSecret . | quote }}
+            {{- end }}
```
Comment (fmulero) [inline bitnami/rabbitmq/templates/statefulset.yaml:370]: Wrong indentation
```suggestion
              {{- if and .Values.advancedConfigurationExistingSecret (not .Values.advancedConfiguration) }}
              - secret:
                  name: {{ tpl .Values.advancedConfigurationExistingSecret . | quote }}
              {{- end }}
```

**Review by fmulero — CHANGES_REQUESTED** [review]
Great!! thanks a lot Francisco for your contribution, it simplifies the implementation, I like it.

There is a little failure, Could you review it?

**Comment by FranAguiar** [comment]
Reviewed the indentation error and also I added the mandatory key in the secret to the values yaml and the readme

**Review by fmulero — APPROVED** [review]
Great!!! LGTM

**Pull request merged**